Lib/argparse.py

```diff
@@ -325,11 +325,19 @@ def _format_usage(self, usage, actions, groups, prefix):
             if len(prefix) + len(usage) > text_width:
 
                 # break usage into wrappable parts
-                part_regexp = r'\(.*?\)+|\[.*?\]+|\S+'
+                part_regexp = '|'.join([
+                    r'\(.*?\)+\s',
+                    r'\[.*?\]+\s',
+                    r'\(.*?\)+$',
+                    r'\[.*?\]+$',
+                    r'\S+',
+                ])
                 opt_usage = format(optionals, groups)
                 pos_usage = format(positionals, groups)
                 opt_parts = _re.findall(part_regexp, opt_usage)
                 pos_parts = _re.findall(part_regexp, pos_usage)
+                opt_parts = [part.strip() for part in opt_parts]
+                pos_parts = [part.strip() for part in pos_parts]
```

Comment:
It would be desirable to be able to avoid these strip calls by tweaking the way that `part_regexp` is defined to look for the trailing whitespace, but not actually capture it. I've added a suggestion above for how to do that with a lookahead assertion.